src/syscheckd/src/ebpf/src/modern.bpf.c

```diff
@@ -399,9 +402,16 @@ int kprobe__security_inode_setattr(struct pt_regs *ctx) {
 SEC("kprobe/vfs_unlink")
 int kprobe__vfs_unlink(struct pt_regs *ctx)
 {
-    struct dentry *dentry = (struct dentry *)PT_REGS_PARM3(ctx);
-    if (!dentry)
-        return 0;
+    struct dentry *dentry;
+    if (LINUX_KERNEL_VERSION >= KERNEL_VERSION(5, 12, 0)) {
+        dentry = (struct dentry *)PT_REGS_PARM3(ctx);
+        if (!dentry) // This condition is necessary to open the BPF program
+            return 0;
+    } else {
+        dentry = (struct dentry *)PT_REGS_PARM2(ctx);
+        if (!dentry) // This condition is necessary to open the BPF program
+            return 0;
+    }
```

Comment:
Why not validate this outside of the if condition? It's required for both cases.


```suggestion
    if (LINUX_KERNEL_VERSION >= KERNEL_VERSION(5, 12, 0)) {
        dentry = (struct dentry *)PT_REGS_PARM3(ctx);
    } else {
        dentry = (struct dentry *)PT_REGS_PARM2(ctx);
    }
    
    if (!dentry) // This condition is necessary to open the BPF program
            return 0;
```